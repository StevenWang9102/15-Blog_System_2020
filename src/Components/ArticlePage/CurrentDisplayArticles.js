import React from "react";
import { Link } from "react-router-dom";
import PropTypes from "prop-types";
import { FavoritedButton } from "../../Components/MainPage/FavoritedButton";
import { ArticleTitle } from "../../Components/ArticlePage/ArticleTitle";

export const CurrentDisplayArticles = props => {

  return (
    <div>
      {props.currentDisplayArticle.map((article, index) => {
    
        return (
          <div className='article-preview' key={index}>
            <div className='article-meta'>
              <ArticleTitle
                pageName={props.pageName}
                article={article}
                setProfileNavStatus={props.setProfileNavStatus}
                loadUserProfileDetail={props.loadUserProfileDetail}
              />

              <FavoritedButton
                httpMethod={props.httpMethod}
                setHttpMethod={props.setHttpMethod}
                article={article}
                userInformation={props.userInformation}
                currentPageOffSet={props.currentPageOffSet}
                onFavoritedArticleClicked={props.onFavoritedArticleClicked}
              />
            </div>

            <Link
              className='nav-link preview-link article-detail'
              to={"/article-detail/" + article.slug}>
              <h1>{article.title}</h1>
              <p>{article.description}</p>
              <span>Read more...</span>
            </Link>
          </div>
        );
      })}

      {props.currentDisplayArticle &&
        props.currentDisplayArticle.length === 0 && (
          <div className='article-preview'>No articles are here... yet.</div>
        )}
    </div>
  );
};

CurrentDisplayArticles.propTypes = {
  currentDisplayArticle: PropTypes.array,
  pageName: PropTypes.string,
  userInformation: PropTypes.object,
  setProfileNavStatus: PropTypes.func,
  loadUserProfileDetail: PropTypes.func,
};