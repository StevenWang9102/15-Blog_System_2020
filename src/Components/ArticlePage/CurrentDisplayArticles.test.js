import React from "react";
import { CurrentDisplayArticles } from "./CurrentDisplayArticles";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <CurrentDisplayArticles
        currentDisplayArticle={[]}
        pageName='test'
        userInformation={{}}
        setProfileNavStatus={() => {}}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
