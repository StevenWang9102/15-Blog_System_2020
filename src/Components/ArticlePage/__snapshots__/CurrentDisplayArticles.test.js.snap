// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="article-preview"
  >
    No articles are here... yet.
  </div>
</div>
`;
